src/IOSPager.tsx

```diff
@@ -0,0 +1,181 @@
+import * as React from 'react';
+import { StyleSheet, Keyboard } from 'react-native';
+import Animated from 'react-native-reanimated';
+
+import { Route, Listener } from './types';
+import { Props } from './Pager';
+
+const { event, divide } = Animated;
+
+type State = {
+  initialOffset: { x: number; y: number };
+};
+
+declare module 'react-native' {
+  export interface ScrollViewProps {
+    scrollToOverflowEnabled?: boolean;
+  }
+}
+
+export default class IOSPager<T extends Route> extends React.Component<
+  Props<T>,
+  State
+> {
+  static defaultProps = {
+    bounces: true,
+  };
+
+  componentDidMount() {
+    if (this.props.layout.width) {
+      this.scrollTo(
+        this.props.navigationState.index * this.props.layout.width,
+        false
+      );
+    }
+  }
+
+  componentDidUpdate(prevProps: Props<T>) {
+    const offset = this.props.navigationState.index * this.props.layout.width;
+
+    if (
+      prevProps.navigationState.routes.length !==
+        this.props.navigationState.routes.length ||
+      prevProps.layout.width !== this.props.layout.width
+    ) {
+      this.scrollTo(offset, false);
+    } else if (
+      prevProps.navigationState.index !== this.props.navigationState.index
+    ) {
+      this.scrollTo(offset);
+    }
+  }
+
+  private initialOffset = {
+    x: this.props.navigationState.index * this.props.layout.width,
+    y: 0,
+  };
+
+  private scrollViewRef: React.RefObject<
+    Animated.ScrollView
+  > = React.createRef();
```

Comment:
`private scrollViewRef = React.createRef<Animated.ScrollView>();` might be enough